### PR TITLE
[Doc] Documentation for distributed inference 

### LINKED ISSUE #206: Documentation on distributed execution

### PR DESCRIPTION
**Pull request by zhuohan123**
In this PR:

- Add anyscale's blogpost to README as a reference for continuous batching.
- Documentation for distributed inference.

Fix #206

### CODE REVIEW

File: docs/source/index.rst [hunk 1]
```diff
@@ -40,7 +40,11 @@ vLLM is flexible and easy to use with:
 * Streaming outputs
 * OpenAI-compatible API server
 
-For more information, please refer to our `blog post <https://vllm.ai>`_.
+For more information:
+
+* `vLLM announcing blog post <https://vllm.ai>`_ (PagedAttention introduction)
+* `Continuous batching introduction <https://www.anyscale.com/blog/continuous-batching-llm-inference>`_ by Cade Daniel et al.
```
Comment (WoosukKwon) [inline docs/source/index.rst:46]: Why don't we just use the original title of the blog post?
```suggestion
* `How continuous batching enables 23x throughput in LLM inference while reducing p50 latency <https://www.anyscale.com/blog/continuous-batching-llm-inference>`_ by Cade Daniel et al.
```

**Review by WoosukKwon — COMMENTED** [review]
Thanks for the PR! Left some comments.

**Comment by zhuohan123** [comment]
@WoosukKwon Just fixed all comments. Please take a look.

**Review by WoosukKwon — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**